Update README.md to include branch matching information (#373)
o3de/o3de-multiplayersample: README.md

<span style="background-color:#4F3C3C">**NOTE:** You can clone the project to any local directory. If you clone the project inside an existing Git repository directory (for example, the directory that contains your local O3DE engine repository) you should add the o3de-multiplayersample project directory to the Git exclude file for the existing Git repository.</span>

> It is expected that o3de, o3de-multiplayersample and o3de-multiplayersample-assets are all cloned from the same branch. If using development o3de-multiplayersample, then match with development branches of o3de, and o3de-multiplayersample. If using a release version of o3de, then checkout the corresponding tags for o3de-multiplayer and o3de-multiplayersample-assets ie `git checkout tags/<tag> -b local branch name>`.

Reviewer: this is very useful info, but I feel like it's too easy to miss up here. Most folks will just jump to the steps (speaking from personal experience, anyway)

what do you think about leaving this but also adding a 5th step to both clone options like: 
"Ensure your o3de, o3de-multiplayersample, and o3de-multiplayersample-assets repos are all checked out on the same branch (such as: `development`). See above note for details."

Reviewer 2: +1 
I'd take it even a little bit further, and just have a step after cloning where you do the git checkout and then reupdate the submodule, and put the description of the choice of branches (development vs stabilization) there.

   echo o3de-multiplayersample-assets > C:/o3de/.git/info/exclude
   ```

## Step 1a. Ensure your branches match

Before building the proeject, ensure that o3de, o3de-multiplayersample and o3de-multiplayersample-assets are all cloned from the same named branches. For example, if you are using the **development** branch of o3de-multiplayersample, then it must be matched with the **development** branches of o3de, and o3de-multiplayersample. Ensure that you update the submodules in o3de-multiplayersample-assets, when switching branches in that repository.

> If you're using an installed version of o3de, then checkout the corresponding versions of the sample reprositories that match the release. O3DE uses tags to identify the release compatible version of each repository. There will be match release tags for o3de-multiplayer and o3de-multiplayersample-assets which can be found via https://github.com/o3de/o3de-multiplayersample/tags. Branches can be checked out using Git's tag syntax, `git checkout tags/<tag> -b local branch name>`.

Author: ```suggestion
If you're using a release or installer version of O3DE, then checkout the corresponding versions of the sample repositories that match the release. O3DE uses [tags](https://git-scm.com/book/en/v2/Git-Basics-Tagging) to identify the release compatible version of each repository. There will be release tags set for o3de-multiplayer and o3de-multiplayersample-assets, which can be found via https://github.com/o3de/o3de-multiplayersample/tags. Branches can be checked out using standard Git commands for example, `git checkout tags/<tag> -b local branch name`.
```

   echo o3de-multiplayersample-assets > C:/o3de/.git/info/exclude
   ```

## Step 1a. Ensure your branches match

> Before building the proeject, ensure that o3de, o3de-multiplayersample and o3de-multiplayersample-assets are all cloned from the same named branches. For example, if you are using the **development** branch of o3de-multiplayersample, then it must be matched with the **development** branches of o3de, and o3de-multiplayersample. Ensure that you update the submodules in o3de-multiplayersample-assets when switching branches in that repository.

Reviewer: ```suggestion
Before building the project, ensure that o3de, o3de-multiplayersample and o3de-multiplayersample-assets are all cloned from the same named branches. For example, if you are using the **development** branch of o3de-multiplayersample, then it must be matched with the **development** branches of o3de, and o3de-multiplayersample-assets. Ensure that you update the submodules in o3de-multiplayersample-assets when switching branches in that repository.
```

> Before building the proeject, ensure that o3de, o3de-multiplayersample and o3de-multiplayersample-assets are all cloned from the same named branches. For example, if you are using the **development** branch of o3de-multiplayersample, then it must be matched with the **development** branches of o3de, and o3de-multiplayersample. Ensure that you update the submodules in o3de-multiplayersample-assets, when switching branches in that repository.

Reviewer 2: ```suggestion
Before building the proeject, ensure that o3de, o3de-multiplayersample and o3de-multiplayersample-assets are all cloned from the same named branches. For example, if you are using the **development** branch of o3de-multiplayersample, then it must be matched with the **development** branches of o3de, and o3de-multiplayersample. Ensure that you update the submodules in o3de-multiplayersample-assets when switching branches in that repository.
```

> If you're using a release or installer version of O3DE, then you must checkout versions of the sample repositories that match the release. O3DE uses standard Git [tags](https://git-scm.com/book/en/v2/Git-Basics-Tagging) to identify the release compatible version of each repository. 

Reviewer 2: ```suggestion
If you're using a release or installer version of O3DE, then you must checkout versions of the sample repositories that match the release. O3DE uses standard Git [tags](https://git-scm.com/book/en/v2/Git-Basics-Tagging) to identify the release-compatible version of each repository. 
```

> For each O3DE release, repositories that have been updated to match the release, should have a matching tag for the release. You can see all defined tags using the [Tags](https://github.com/o3de/o3de-multiplayersample/tags] view in each repository.

Reviewer 2: ```suggestion
For each O3DE release, repositories that have been updated to match the release should have a matching tag for the release. You can see all defined tags using the [Tags](https://github.com/o3de/o3de-multiplayersample/tags] view in each repository.
```
